### DISCUSSION TITLE
Firebase? Yay or nay?

### DISCUSSION THREAD

**Discussion by redstone-dev** [SnarpleDev/snazzle-sk#2] [Polls]
Should we use Firebase for our backend?

**Comment by mybearworld** [comment]
Hm, why does this need a backend?

**Reply by redstone-dev** [reply]
For storing user preferences such as themes and other options. Really, all we need is a database, so actually SQLite might be better :P

**Reply by mybearworld** [reply]
localStorage?

**Reply by redstone-dev** [reply]
Oh my god, I totally forgot about localStorage 🤦🏻 
Thanks for that!

**Reply by redstone-dev** [reply]
Turns out that SvelteKit hates client side stuff. Also, we need a place to store feedback so we can view it. (there's a page where you can give feedback) @mybearworld

**Reply by mybearworld** [reply]
"a place to store feedback so we can view it" probably would be unmoderated chat then, though. Also, what do you mean SvelteKit hates client side stuff?

**Reply by NotFenixio** [reply]
> Also, what do you mean SvelteKit hates client side stuff?

Check the answer from https://stackoverflow.com/questions/75770128/how-to-retrieve-localstorage-data-on-a-sveltekit-page-js-file

**Reply by mybearworld** [reply]
Oh, yeah. https://kit.svelte.dev/docs/modules#$app-environment-browser is a solution, though.

**Reply by redstone-dev** [reply]
@mybearworld it wouldn't be unmoderated chat because it would be directly with the admins, and also, you can disable ssr but it wouldn't work for me, but I realize now what the fix was

**Comment by davidtheplatform** [comment]
1. Can you see who voted what?
2. If we use firebase it would be really easy to accidentally push the API key

**Reply by NotFenixio** [reply]
I believe that to address this issue, we could maintain a private repository containing our implementation for the main Snazzle server, and another repository for the self-deployable version.
